specs/experimental/plasma.md

```diff
@@ -83,6 +83,16 @@ on the `commitment_type_byte` where [0, 128) are reserved for official implement
 | `commitment_type` | `commitment`                    |
 | ----------------- | ------------------------------- |
 | 0                 | `keccak256(tx_payload)`         |
+| 1                 | `da-service`                     |
+
+The `da-service` commitment is as follows: `da_layer_byte ++ len(payload) as uvarint ++ payload`. The DA layer byte
+must be initially restricted to the range `[0, 127)`. This specification will not apportion DA layer bytes, but
+different DA layers should coordinate to ensure that the DA layer bytes do not conflict. The payload length is a
+unsigned 64 bit number, as defined in [protobuf spec]. The payload is a bytestring which is up to the DA layer to
+specify. The DA server should be able to parse the payload, find the data on the DA server, & verify that the data
+returned from the DA server matches what was committed to in the payload.
```

Comment:
Would this handled similarly in the `DataAvailabilityChallenge` contract? It would be ideal if the challenge resolution can be left to the implementation as well, because we can have compact proofs instead of having to submit the entire blob.

Reply:
👍 
It would be nice since there are constructions where proving data unavailability can be done without sending the actual blob. Also, proving invalid state transitions can be reduced to providing a single share. 
So, it would be good if the resolution is implementation specific

Reply:
We do want to let the DA commitment resolution be handled by the contract, but this is focusing on just the commitment <-> da server portion right now.